client/my-sites/site-settings/cloudflare-analytics/index.js

```diff
@@ -0,0 +1,198 @@
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { flowRight, partialRight, pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wrapSettingsForm from '../wrap-settings-form';
+import { Card } from '@automattic/components';
+import { isDesktop } from '@automattic/viewport';
+import { getPlugins } from 'calypso/state/plugins/installed/selectors';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextValidation from 'calypso/components/forms/form-input-validation';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import getCurrentRouteParameterized from 'calypso/state/selectors/get-current-route-parameterized';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import SettingsSectionHeader from 'calypso/my-sites/site-settings/settings-section-header';
+import { localizeUrl } from 'calypso/lib/i18n-utils';
+import cloudflareIllustration from 'calypso/assets/images/illustrations/cloudflare-logo.svg';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const validateTrackingId = ( code ) => ! code || code.match( /^(UA-\d+-\d+)|(G-[A-Z0-9]+)$/i ); // @TODO: how are cloudflare tracking IDs formatted?
+
+export class CloudflareAnalyticsSettings extends Component {
+	state = {
+		isCodeValid: true,
+		loggedCloudflareAnalyticsModified: false,
+	};
+
+	handleFieldChange = ( key, value ) => {
+		const { fields, updateFields } = this.props;
+		const updatedCloudflareFields = Object.assign( {}, fields.cloudflare || {}, {
+			[ key ]: value,
+		} );
+		updateFields( { cloudflare: updatedCloudflareFields } );
+	};
+
+	handleCodeChange = ( event ) => {
+		const code = event.target.value.trim();
+
+		this.setState( {
+			isCodeValid: validateTrackingId( code ),
+		} );
+		this.handleFieldChange( 'code', code );
+	};
+
+	isSubmitButtonDisabled() {
+		const { isRequestingSettings, isSavingSettings } = this.props;
+		return (
+			isRequestingSettings ||
+			isSavingSettings ||
+			! this.state.isCodeValid ||
+			! this.props.enableForm
+		);
+	}
+
+	form() {
+		const {
+			enableForm,
+			eventTracker,
+			fields,
+			handleSubmitForm,
+			isRequestingSettings,
+			isSavingSettings,
+			path,
+			translate,
+			trackTracksEvent,
+			uniqueEventTracker,
+		} = this.props;
+		const placeholderText = isRequestingSettings ? translate( 'Loading' ) : '';
+		const analyticsSupportUrl = 'https://wordpress.com/support/CLOUDFLARE_SUPPORT_URL/'; // TODO: add support link
+
+		return (
+			<form id="analytics" onSubmit={ handleSubmitForm }>
+				<SettingsSectionHeader
+					disabled={ this.isSubmitButtonDisabled() }
+					isSaving={ isSavingSettings }
+					onButtonClick={ handleSubmitForm }
+					showButton={ true }
```

Comment:
Shorthand doesn't need the "true", but feel free to leave it if you prefer to be explicit